whylog/log_reader/searchers.py

```diff
@@ -42,6 +42,12 @@ def _deduce_offset(self, investigation_step):
             if investigation_step.is_line_in_search_range(line_content):
                 return line_offset + len(line_content) + 1
 
+    def _find_offset_by_date(self, opened_file, date):
```

Comment:
In my opinion this method has to change signature. I planed that this method takes opened_file, primary key value with represents left or right bound and super_parser which can extract primary value from almost every line in opened_file.